src/System.IO.FileSystem/tests/Directory/CreateDirectory.cs

```diff
@@ -206,19 +206,6 @@ public void DirectoryEqualToMaxDirectory_CanBeCreatedAllAtOnce()
             Assert.True(Directory.Exists(result.FullName));
         }
 
-        [Fact]
-        public void DirectoryWithComponentLongerThanMaxComponentAsPath_ThrowsPathTooLongException()
-        {
-            // While paths themselves can be up to 260 characters including trailing null, file systems
-            // limit each components of the path to a total of 255 characters.
-            var paths = IOInputs.GetPathsWithComponentLongerThanMaxComponent();
-
-            Assert.All(paths, (path) =>
-            {
-                Assert.Throws<PathTooLongException>(() => Create(path));
-            });
-        }
-
```

Comment:
> The exception should be compatible with Desktop for compat.

It can't be by design.

Windows doesn't always return "TooLong". The whole point of this is to not second-guess the OS. We already get this wrong for some file system formats and we're currently blocking any new ones that might support > 255.